### PR TITLE
Ensure attach lifecycle callback is not triggered if it is not in the DOM.

### PR DESCRIPTION
**Pull request by treshugart**

### CODE REVIEW

File: src/lifecycle.js [hunk 1]
```diff
@@ -269,21 +253,22 @@ function addEventListeners (target, component) {
...
+  targetData.set('created', true);
+
   // TODO: This doesn't need to happen if using native.
   inherit(target, component.prototype, true);
 
   // We use the unresolved / resolved attributes to flag whether or not the
-  // element has been templated or not. The resolved attribute is also how
-  // you can ensure there isn't a FOUC. If no templating is happening, then
-  // there is no reason to add / remove these attributes.
-  if (component.template && !target.hasAttribute(component.resolvedAttribute)) {
```
Comment (nicolasv) [inline src/lifecycle.js]: Why'd you move away from using `resolvedAttribute`? Does this mean that #133 (prerendering things server-side) should be reopened?

File: test/unit/init.js [hunk 2]
```diff
@@ -153,16 +153,12 @@ describe('Instantiation', function () {
...
+          idsToCheck.push(id);
         }
       });
     });
 
     skate.init(div);
-
-    idsToSkate.forEach(function (id) {
-      idsToCheck[id].should.equal(true);
-    });
```
Comment (nicolasv) [inline test/unit/init.js]: What assertion is made in this test now (if any)?

File: test/unit/lifecycle.js [hunk 3]
```diff
@@ -4,38 +4,66 @@ import helpers from '../lib/helpers';
 import skate from '../../src/skate';
 
 describe('Lifecycle Callbacks', function () {
-  it('should call the created() callback when the element is attached', function (done) {
-    var tagName = helpers.safeTagName('my-element');
-    skate(tagName.safe, {
+  var MyEl;
+  var myEl;
```
Comment (nicolasv) [inline test/unit/lifecycle.js]: Hmm, I'm generally not a fan of var names that are this similar. Would you consider `myEl` and `myElConstructor` (or `myElCtor`)?

Comment (treshugart) [inline test/unit/lifecycle.js]: Sure.

File: src/lifecycle.js (hunk 1 above)
Comment (treshugart) [inline src/lifecycle.js]: I'm working on a separate commit to re-address this.

File: test/unit/init.js (hunk 2 above)
Comment (treshugart) [inline test/unit/init.js]: I should probably actually remove this test. It's hard to grok and it's not testing anything useful. The original problem was specific to firefox and existed when you had a component called `watch`. This isn't a problem if using the recommended spec'd naming convention which we're recommending moving forward.

**Pull request merged**